doc/reference/std/misc/atom.md

```diff
@@ -0,0 +1,339 @@
+```
+
+Returns `#t` if *a* an atom, `#f` otherwise.
+
+::: tip Examples:
+``` scheme
+> (atom? (atom))
+t
+
+> (atom 42)
+#<Atom #8>
+> (atom? #8)
+#t
+```
+:::
+
+## atom-swap!
+``` scheme
+(atom-swap! atom function) -> any
+```
+
+Applies the function to the value currently in the atom, and,
+if it passes the atom's validator, sets the atom to the new value.
+Ensures that no other access is made to the atom in the middle of a swap.
+Returns the new value.
+
+::: tip Examples:
+``` scheme
+> (def a (atom 3))
+> (atom-swap! a (cut * 2 <>))
+6
+```
+:::
+
+## atom-swap-values!
+``` scheme
+(atom-swap! atom function) -> any any
+```
+
+Similar to `atom-swap!` but returns two values, the `old` and the `new` values.
+Similar to Clojure's `swap-vals!` but returns two values rather than a vector of size 2.
+
+::: tip Examples:
+``` scheme
+> (def a (atom 6))
+> (atom-swap-values! a (cut * 2 <>))
+6
+12
+```
+:::
+
+## atom-reset!
+``` scheme
+(atom-reset! atom new-value) -> new-value
+```
+
+If the new value passes the atom's validator, sets the atom to it.
+Ensures that no other access is made to the atom in the middle of a reset.
+Returns the new value.
+
+::: tip Examples:
+``` scheme
+> (def a (atom 42))
+> (atom-reset! a 23)
+23
+> (atom-deref a)
+23
+```
+:::
+
+## atom-reset-values!
+``` scheme
+(atom-reset-values! atom new-value) -> old-value new-value
+```
+
+Similar to `atom-reset!` but returns two values, the old and the new.
+Similar to Clojure `atom-reset-vals!` but returns values instead of a vector.
+
+::: tip Examples:
+``` scheme
+> (def a (atom 42))
+> (atom-reset-values! a 23)
+42
+23
+> (atom-deref a)
+23
+```
+:::
+
+## atom-compare-and-set!
+
+``` scheme
+(atom-compare-and-set! atom oldval newval) -> bool
+```
+
+Atomically sets the value of atom to newval if and only if the
+current value of the atom is identical to oldval.
+Returns `#t` if set happened, else `#f`.
+Similar to Clojure `compare-and-set!`.
+
+::: tip Examples:
+``` scheme
+> (def a (atom 42))
+> (atom-compare-and-set! a 42 23)
+#t
+> (atom-deref a)
+23
+> (atom-compare-and-set! a 42 24)
+#f
+> (atom-deref a)
+23
+```
+:::
+
+## atom-validator
+
+``` scheme
+(atom-validator atom) -> validator
+```
+
+Access the current atom's validator, a function that takes a new value as argument
+and returns a true value iff the atom will accept that value.
+The value `#f` is a stand-in for not doing any validation,
+which is equivalent to the function `true` that accepts every value.
+Similar to Clojure `get-validator`.
+
+::: tip Examples:
+``` scheme
+> (def a (atom 42 validator: number?))
+> (atom-reset! a 1)
+1
+> (atom-reset! a "foo")
+*** ERROR IN with-lock --
+*** ERROR IN "misc/atom.ss"@48.29 [BadArgument]: bad argument; expected valid atom value
+--- irritants: update-atom! "foo"
+```
+:::
+
+## atom-validator-set!
+
+``` scheme
+(atom-validator-set! atom validator) -> _
+```
+
+Modify the current atom's validator, a function that takes a new value as argument
```

Comment:
Modify `atom`'s validator. The current value is grandfathered and not validated.
The `validator` is not validated for whether it accepts the current value.
The function is not guaranteed to return any specific value.
Similar to Clojure `set-validator!`.